tests: various/plugin.sh fails when PREFIX != /usr/local/share

This has already been brought up in https://github.com/YosysHQ/yosys/commit/f11dd6e20838bce1b1fa8e085570d2126484d10f#r38433102, but I'll add it here for better visibility and tracking.

## Steps to reproduce the issue

```
make -j8 PREFIX=/usr/share test
```

## Expected behavior

The tests complete without error.

## Actual behavior

`tests/various/plugin.cc` fails to compile with:

```
plugin.cc:1:10: fatal error: 'kernel/rtlil.h' file not found
#include "kernel/rtlil.h"
         ^~~~~~~~~~~~~~~~
```

This is due to the replacement of include path flags that happens in `plugin.sh`, which assumes `PREFIX=/usr/local/share`.

--- a/misc/yosys-config.in
+++ b/misc/yosys-config.in
@@ -6,7 +6,7 @@ help() {
 		echo "Usage: $0 [--exec] [--prefix pf] args.."
 		echo "       $0 --build modname.so cppsources.."
 		echo ""
-		echo "Replecement args:"
+		echo "Replacement args:"
 		echo "    --cxx         @CXX@"
 		echo "    --cxxflags    $( echo '@CXXFLAGS@' | fmt -w60 | sed ':a;N;$!ba;s/\n/ \\\n                      /g' )"
 		echo "    --ldflags     @LDFLAGS@"

--- a/tests/various/plugin.sh
+++ b/tests/various/plugin.sh
@@ -1,6 +1,8 @@
 set -e
 rm -f plugin.so
 CXXFLAGS=$(../../yosys-config --cxxflags)
-CXXFLAGS=${CXXFLAGS// -I\/usr\/local\/share\/yosys\/include/ -I..\/..\/share\/include}
+DATDIR=$(../../yosys-config --datdir)
+DATDIR=${DATDIR//\//\\\/}
+CXXFLAGS=${CXXFLAGS//$DATDIR/..\/..\/share}
 ../../yosys-config --exec --cxx ${CXXFLAGS} --ldflags -shared -o plugin.so plugin.cc
 ../../yosys -m ./plugin.so -p "test" | grep -q "Plugin test passed!"

